app/src/vesselInfo/vesselInfoActions.js

```diff
@@ -265,48 +254,23 @@ export const applyFleetOverrides = () => (dispatch, getState) => {
 export function setPinnedVessels(pinnedVessels, shownVessel) {
   return (dispatch, getState) => {
     const state = getState();
-    const options = {
-      Accept: '*/*'
-    };
-
-    options.headers = {
-      Authorization: `Bearer ${state.user.token}`
-    };
+    const { token } = state.user;
 
     pinnedVessels.forEach((pinnedVessel) => {
-      let request;
-
-      if (typeof XMLHttpRequest !== 'undefined') {
-        request = new XMLHttpRequest();
-      } else {
-        throw new Error('XMLHttpRequest is disabled');
-      }
-
       const layer = state.layers.workspaceLayers.find(l => l.tilesetId === pinnedVessel.tilesetId);
       if (layer === undefined) {
         console.warn('Trying to load a pinned vessel but the layer seems to be absent on the workspace', pinnedVessel);
         return;
       }
 
-      request.open(
-        'GET',
-        buildEndpoint(layer.header.endpoints.info, { id: pinnedVessel.seriesgroup }),
-        true
-      );
-      if (state.user.token) {
-        request.setRequestHeader('Authorization', `Bearer ${state.user.token}`);
-      }
-      request.setRequestHeader('Accept', 'application/json');
-      request.onreadystatechange = () => {
-        if (request.readyState !== 4) {
-          return;
-        }
-        if (request.status === 404) {
-          console.warn('Error:', request.responseText);
-          return;
-        }
-        const data = JSON.parse(request.responseText);
+      const pinnedVesselUrl = buildEndpoint(layer.header.endpoints.info, { id: pinnedVessel.seriesgroup });
+      fetchEndpoint(pinnedVesselUrl, token).then((data) => {
         delete data.series;
+        // mocking API comments to show warning
+        // TODO:
+        //  1. Check field name from API
+        //  2. Remove the next line of code
+        data.comment = 'This vessel has a warning PINNED';
```

Comment:
@j8seangel I would remove this now so we can merge the PR, once the API starts delivering the comment it should work.